app/code/Magento/CatalogGraphQl/Model/Resolver/Products/DataProvider/ProductSearch.php

```diff
@@ -110,7 +110,7 @@ public function getList(
         $searchResults = $this->searchResultsFactory->create();
         $searchResults->setSearchCriteria($searchCriteriaForCollection);
         $searchResults->setItems($collection->getItems());
-        $searchResults->setTotalCount($searchResult->getTotalCount());
+        $searchResults->setTotalCount($collection->getSize());
```

Comment:
Please check this change. It may need to be reverted, it seems to be the cause of web api test failure

Reply:
I investigated this a bit. This change is causing the web api test failure. But there is another issue mixed up with it.
So I think we should not make this change in your PR. Instead we will address the other issue separately, and your builds will be green

Reply:
reverted this.